### PR TITLE
Simplify configuration

### PR DESCRIPTION
**Pull request by ylavoie**
- Simplify the ledgersmb/Dockerfile so that we can select previous branches to include
- Fix pgdata location to enable PG configuration
- Build dojo at runtime instead of build time
- Create the HOME directory to prevent docker-compose to create it on first mount and assign it to root

### CODE REVIEW

File: ledgersmb/start.sh [hunk 1]
```diff
@@ -27,6 +27,8 @@ fi
 PERL5LIB=lib
 export PERL5LIB
 
+make dojo
```
Comment (ehuelsmann) [inline ledgersmb/start.sh:30]: Please make this conditional on the build not having been executed before: I regularly need to restart the container after i change Perl code; when i change JS code, the container doesn't need restarting... So this runs `make dojo` exactly when I'm likely not to need it.

File: postgres/lsmb-dev-init.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 
-pgdata=${PGDATA:-/var/lib/postgresql/data}
+pgdata=${PGDATA:-/var/lib/postgresql/data/pgdata}
```
Comment (ehuelsmann) [inline postgres/lsmb-dev-init.sh:4]: Why this change? PostgreSQL explicitly documents that it's data directory should *not* be a mountpoint which is why we have this directory in place the way it is now.

Or are you saying that the bit below doesn't work? Then why does my container log all queries?

File: ledgersmb/Dockerfile [hunk 3]
```diff
@@ -108,62 +113,30 @@ RUN set -x ; \
...
-  git remote update && \
-  eatmydata cpanm --quiet --notest \
+  git remote update
+
+# FROM resets ARG, so redeclare
+ARG branches
+
+RUN cd /srv/ledgersmb && \
+  bash -c 'for branch in ${branches//,/ }; do \
+    cd /srv/ledgersmb && \
```
Comment (ehuelsmann) [inline ledgersmb/Dockerfile:123]: Very nice cleanup!

File: ledgersmb/Dockerfile [hunk 4]
```diff
@@ -1,3 +1,5 @@
+ARG branches="1.6,1.7,1.8,1.9,master"
```
Comment (ehuelsmann) [inline ledgersmb/Dockerfile:1]: Very nice idea!

However, this drops 1.5 support in the container. Is that accidental?

**Review by ehuelsmann — COMMENTED** [review]
Very nice improvements over all. Just a few questions before we should proceed.

File: ledgersmb/Dockerfile (hunk 4 above)
Comment (ylavoie) [reply ledgersmb/Dockerfile:1]: I seems to remember that we discussed removing it.

File: postgres/lsmb-dev-init.sh (hunk 2 above)
Comment (ylavoie) [reply postgres/lsmb-dev-init.sh:4]: postgresql.conf is in `/var/lib/postgresql/data/pgdata`, not `/var/lib/postgresql/data`

File: ledgersmb/start.sh (hunk 1 above)
Comment (ylavoie) [reply ledgersmb/start.sh]: Done

File: postgres/lsmb-dev-init.sh (hunk 2 above)
Comment (ylavoie) [reply postgres/lsmb-dev-init.sh:4]: Yours work because the docker.compose.yml file sets PGDATA properly

File: ledgersmb/Dockerfile (hunk 3 above)
Comment (ylavoie) [reply ledgersmb/Dockerfile:123]: Thanks

File: ledgersmb/Dockerfile (hunk 4 above)
Comment (ehuelsmann) [reply ledgersmb/Dockerfile:1]: Hmm. I may be inconsistent, but i prefer to keep 1.5 as long as feasible.

File: postgres/lsmb-dev-init.sh (hunk 2 above)
Comment (ehuelsmann) [reply postgres/lsmb-dev-init.sh:4]: Ah! Thanks for that clarification.
